src/app-layer-tftp.c

```diff
@@ -0,0 +1,345 @@
+/* Copyright (C) 2015 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ */
+
+/**
+ * \file
+ *
+ * \author Clément Galland <clement.galland@epita.fr>
+ *
+ * TFTP application layer detector and parser for learning and
+ * tftp pruposes.
+ *
+ * This tftp implements a simple application layer for something
+ * like the echo protocol running on port 7.
+ */
+
+
+#include "suricata-common.h"
+#include "stream.h"
+#include "conf.h"
+
+#include "util-unittest.h"
+
+#include "app-layer-detect-proto.h"
+#include "app-layer-parser.h"
+
+#include "app-layer-tftp.h"
+
+#ifndef HAVE_RUST
+
+void RegisterTFTPParsers(void)
+{
+}
+
+#else
+
+#include "rust-tftp-tftp-gen.h"
+
+/* The default port to probe for echo traffic if not provided in the
```

Comment:
Please get rid of echo protocol references.

Reply:
agreed.